text/0000-tracked-properties.md

```diff
@@ -227,34 +248,350 @@ memoization) if a property shows up as a bottleneck during profiling.
 
 ### Native Classes
 
-Tracked properties are designed from the ground up for native JavaScript (ES6) classes.
-
-### "Just JavaScript"
-
+Tracked properties are designed from the ground up for native JavaScript (ES6)
+classes. Adopting them will make it easier to convert away from the aging Ember
+object-model, and
 
 ## Detailed design
 
 ### Autotracking Stack
-> This is the bulk of the RFC.
 
-> Explain the design in enough detail for somebody
-familiar with the framework to understand, and for somebody familiar with the
-implementation to implement. This should get into specifics and corner-cases,
-and include examples of how the feature is used. Any new terminology should be
-defined here.
+Autotracking works by taking advantage of the fact that all mutable properties
+will be decorated by `tracked`. This decorator adds a getter and a setter that
+wraps the property, be it a tracked getter or a tracked simple property. Every
+time we activate a tracked getter, it creates a global stack before it begins
+executing the getter's code, and as it executes, any other tracked wrappers
+that run add themselves to the stack.
+
+This example demonstrates how this method works, but is simplified to only work
+with tracked getters, and to only track keys. It uses the stage 2 decorator
```

Comment:
Not sure if enough people are familiar with decorators (and specifically how you implement a decorator) to make this example improve overall clarity? Maybe there's some pseudocode or a simplified example using just functions? I want to understand the takeaway people should have from this example that they don't get from the prose.

Reply:
I think I can redo this using just `Object.defineProperty` manually. The main thing is that this is a tricky concept, it uses a form of recursion, kind of, and private global state to keep track of things. I think this is hard to describe in prose for people who are unfamiliar with the concepts, and a solid example with working code really will help to clarify for those readers